fvm/transactionVerifier.go

```diff
@@ -14,6 +16,123 @@ import (
 	"github.com/onflow/flow-go/module/trace"
 )
 
+type signatureType struct {
+	message []byte
+
+	errorBuilder func(flow.TransactionSignature, error) errors.CodedError
+
+	aggregateWeights map[flow.Address]int
+}
+
+type signatureEntry struct {
+	flow.TransactionSignature
+
+	signatureType
+
+	accountKey flow.AccountPublicKey
+
+	verifyErr errors.CodedError
```

Comment:
I'm a bit concerned about this including the verifyErr in this data structure because there are two different cases we can't distinguish where verifyErr are both `nil`:

1. we have verified and the signature is valid
2. we haven't verified yet

In other words, if we forgot to call `verify()`, then the signature would seem to be verified and is valid.

It's better to add another field: `verified`, so that we can distinguish them more explicitly: for case 1, `verified` would be `true`; for case 2, `verified` would be `false`.

Reply:
see my continuation state comment